src/twisted/names/dns.py

```diff
@@ -378,7 +379,36 @@ def _isSubdomainOf(descendantName, ancestorName):
     return descendantLabels[-len(ancestorLabels) :] == ancestorLabels
 
 
-def str2time(s):
+def _str2time(s: str) -> int:
+    suffixes = (
+        ("S", 1),
+        ("M", 60),
+        ("H", 60 * 60),
+        ("D", 60 * 60 * 24),
+        ("W", 60 * 60 * 24 * 7),
+        ("Y", 60 * 60 * 24 * 365),
+    )
+    s = s.upper().strip()
+    for (suff, mult) in suffixes:
+        if s.endswith(suff):
+            return int(float(s[:-1]) * mult)
+    try:
+        return int(s)
+    except ValueError:
+        raise ValueError("Invalid time interval specifier: " + s)
+
+
+@overload
+def str2time(s: Union[str, bytes, int]) -> int:
+    ...
+
+
+@overload
+def str2time(s: None) -> None:
```

Comment:
str2time can technically pass through anything but int and None are the only usefull ttl passthrough